src/providers/tenderly-simulation-provider.ts

```diff
@@ -399,91 +399,131 @@ export class TenderlySimulator extends Simulator {
+          MetricLoggerUnit.Milliseconds
+        );
+        metric.putMetric(
+          `TenderlySimulationUniversalRouterResponseStatus${httpStatus}`,
+          1,
+          MetricLoggerUnit.Count
+        );
 
-      const latencies = Date.now() - before;
-      log.info(
-        `Tenderly simulation universal router request body: ${JSON.stringify(
-          body,
-          null,
-          2
-        )} with result ${JSON.stringify(
-          resp,
-          null,
-          2
-        )}, having latencies ${latencies} in milliseconds.`
-      );
-      metric.putMetric(
-        'TenderlySimulationUniversalRouterLatencies',
-        Date.now() - before,
-        MetricLoggerUnit.Milliseconds
-      );
-      metric.putMetric(
-        `TenderlySimulationUniversalRouterResponseStatus${httpStatus}`,
-        1,
-        MetricLoggerUnit.Count
-      );
+        // Validate tenderly response body
+        if (
+          !resp ||
+          !resp.result ||
+          resp.result.length < 3 ||
+          (resp.result[2] as JsonRpcError).error
+        ) {
+          log.error(
+            { resp },
+            `Failed to invoke Tenderly Node Endpoint for gas estimation bundle ${JSON.stringify(
+              body,
+              null,
+              2
+            )}.`
+          );
+          return { ...swapRoute, simulationStatus: SimulationStatus.Failed };
+        }
 
-      // Validate tenderly response body
-      if (
-        !resp ||
-        resp.simulation_results.length < 3 ||
-        !resp.simulation_results[2].transaction ||
-        resp.simulation_results[2].transaction.error_message
-      ) {
-        this.logTenderlyErrorResponse(resp);
-        return { ...swapRoute, simulationStatus: SimulationStatus.Failed };
-      }
+        // Parse the gas used in the simulation response object, and then pad it so that we overestimate.
+        estimatedGasUsed = BigNumber.from(
+          (
+            Number((resp.result[2] as GasBody).gas) * estimateMultiplier
+          ).toFixed(0)
+        );
 
-      // Parse the gas used in the simulation response object, and then pad it so that we overestimate.
-      estimatedGasUsed = BigNumber.from(
-        (
-          resp.simulation_results[2].transaction.gas * estimateMultiplier
-        ).toFixed(0)
-      );
+        log.info(
+          {
+            body,
+            approvePermit2GasUsed:
+            (resp.result[0] as GasBody).gasUsed,
+            approveUniversalRouterGasUsed:
+            (resp.result[1] as GasBody).gasUsed,
+            swapGasUsed: (resp.result[2] as GasBody).gasUsed,
+            approvePermit2Gas: (resp.result[0] as GasBody).gas,
+            approveUniversalRouterGas: (resp.result[1] as GasBody).gas,
+            swapGas: (resp.result[2] as GasBody).gas,
+            swapWithMultiplier: estimatedGasUsed.toString(),
+          },
+          'Successfully Simulated Approvals + Swap via Tenderly for Universal Router. Gas used.'
```

Comment:
lets add something tos how its the api or not

Reply:
so I differentiate via the logging `tenderly node endpoint` vs `tenderly api endpoint`. In the meanwhile, I also added endpoint specific metrics `Tenderly[Node|Api]SimulationUniversalRouterLatencies` and `Tenderly[Node|Api]SimulationUniversalRouterResponseStatus`. Eventually as we migrate to node endpoint at 100%, the original metrics `TenderlySimulationUniversalRouterLatencies` and `TenderlySimulationUniversalRouterResponseStatus` should just work as is.